From 7cc7443138f2cd2c9b7dbfc06e908476fa797532 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Thu, 19 Dec 2019 14:22:32 +0800
Subject: [PATCH 07/12] media: Enable mojo media when using v4l2 codec on
 desktop linux

The newest chromium turns to use mojo video decoder instead of gpu video
decoder.

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 src/3rdparty/chromium/media/media_options.gni | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/3rdparty/chromium/media/media_options.gni b/src/3rdparty/chromium/media/media_options.gni
index 96cc9af99..1ff30ed57 100644
--- a/src/3rdparty/chromium/media/media_options.gni
+++ b/src/3rdparty/chromium/media/media_options.gni
@@ -158,7 +158,7 @@ declare_args() {
   enable_mojo_media =
       !is_fuchsia &&
       (is_android || is_chromeos || is_mac || is_win || enable_library_cdms ||
-       (is_desktop_linux && use_vaapi) || is_chromecast)
+       (is_desktop_linux && (use_vaapi || use_v4l2_codec)) || is_chromecast)
 
   # Enable the TestMojoMediaClient to be used in mojo MediaService. This is for
   # testing only and will override the default platform MojoMediaClient, if any.
@@ -223,7 +223,7 @@ if (enable_mojo_media) {
     ]
     _default_mojo_media_host = "gpu"
   } else if (is_chromeos || is_mac || is_win ||
-             (is_desktop_linux && use_vaapi)) {
+             (is_desktop_linux && (use_vaapi || use_v4l2_codec))) {
     _default_mojo_media_services = [ "video_decoder" ]
     _default_mojo_media_host = "gpu"
   }
-- 
2.20.1

